### PR TITLE
marginnote 4.1.4

### PR DESCRIPTION
**Pull request by foreverangelwing**
### Description

This pull request updates the Homebrew Cask for MarginNote from version 3.7.25 to version 4.14.

### Changes:
- Updated `version` to `4.14`
- Updated `url` to `https://d.marginnote.cn/MarginNote4.dmg`
- Updated `app` target to `MarginNote 4.app`
- Updated `sha256` to the new checksum for the updated dmg file
- Updated the `livecheck` URL to `https://dist.marginnote.cn/marginnote4.xml`

### Reason for the Update:
MarginNote has released version 4.14 with new features and improvements. This update reflects the latest available version to ensure users can download and use the most current release.

Let me know if further modifications or corrections are needed. Thanks for reviewing!

### CODE REVIEW

File: Casks/m/marginnote.rb [hunk 1]
```diff
@@ -1,22 +1,22 @@
 cask "marginnote" do
-  version "3.7.25"
-  sha256 "4409e29b6046fc71442f3f2085375dada87b3f06a90892b2334823b0dad75ad3"
+  version "4.14"
```
Comment (khipp) [inline Casks/m/marginnote.rb:2]:
```suggestion
  version "4.1.4"
```

File: Casks/m/marginnote.rb [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
+  sha256 "0bea1a7d4c9a27b03045a4196561877c8fa47e9760250ea09afde80550ef5f4d"
 
-  url "https://marginstudy.com/mac/MarginNote#{version.major}.dmg",
-      verified: "marginstudy.com/mac/"
+  url "https://d.marginnote.cn/MarginNote4.dmg",
```
Comment (khipp) [inline Casks/m/marginnote.rb:5]:
```suggestion
  url "https://dist.marginnote.cn/MarginNote#{version}.dmg",
```

File: Casks/m/marginnote.rb [hunk 3]
```diff
@@ -1,22 +1,22 @@
...
+      verified: "marginnote.cn/"
   name "MarginNote"
   desc "E-reader"
   homepage "https://www.marginnote.com/"
 
   livecheck do
-    url "https://dist.marginnote.cn/marginnote#{version.major}.xml"
+    url "https://dist.marginnote.cn/marginnote4.xml"
```
Comment (khipp) [inline Casks/m/marginnote.rb:12]:
```suggestion
    url "https://dist.marginnote.cn/marginnote#{version.major}.xml"
```

File: Casks/m/marginnote.rb [hunk 4]
```diff
@@ -1,22 +1,22 @@
...
     strategy :sparkle, &:short_version
   end
 
   auto_updates true
-  depends_on macos: ">= :sierra"
+  depends_on macos: ">= :sequoia"
```
Comment (khipp) [inline Casks/m/marginnote.rb:17]:
```suggestion
  depends_on macos: ">= :monterey"
```

File: Casks/m/marginnote.rb [hunk 5]
```diff
@@ -1,22 +1,22 @@
...
 
-  app "MarginNote #{version.major}.app"
+  app "MarginNote 4.app"
```
Comment (khipp) [inline Casks/m/marginnote.rb:19]:
```suggestion
  app "MarginNote #{version.major}.app"
```

**Review by khipp — COMMENTED** [review]
Thank you for your first contribution to `homebrew-cask`, @foreverangelwing! 🎉

There are still a few minor issues to address:

We should continue using interpolation for the version rather than hardcoding it in other stanzas. The placeholder `#{version.major}` will be replaced by the major version (e.g., 4) during evaluation.

Please also squash your commits to maintain a cleaner history.

**Review by bevanjkay — APPROVED** [review]

**Pull request merged**